### DISCUSSION TITLE
Set Style for Individual Feed or Group

### DISCUSSION THREAD

**Discussion by Habay** [ViennaRSS/vienna-rss#62] [Ideas]
Set a style for one feed or for a group of feeds, overruling the default style. 

This could be done from the info window with an option 'Overrule default style' where one could select the desired style from a drop down menu (which hosts the same styles as found in the View->Style menu). 

As groups dont have an info window yet, a group-info window should be added with the same option.

**Comment by barijaona** [comment]
I am afraid I don't understand the request - or more precisely, what motivates it.

Would you please provide a sample of feeds/styles combination that you think might be useful ?

Personally, I would be more interested with a layout per feed : I like the Report layout for feeds delivering only short excerpts of the article, and the Unified layout with feeds giving the whole article.

**Comment by Habay** [comment]
> Personally, I would be more interested with a layout per feed : I like the 
> Report layout for feeds delivering only short excerpts of the article, and the Unified 
> layout with feeds giving the whole article.

Yes, this is what i meant with the 'Overrule default style' option in the feed info window; to be able to select a different layout for certain types of feeds.

But i see good use to bee able to that in one go for group of similar feeds too. I.e. to be able to assign the same layout for a group of newsfeeds and another layout for a group of usenet feeds, etc.

**Comment by rcaputo** [comment]
For example, MacUpdate.  My usage pattern is to browse the headers and not really read the content.  I think the ability to eliminate the content pane entirely (for this one subscription) would improve the experience.

I'd also like to override the default sort order to always sort by Subject.  That way I can skip through multiple updates for the same program.  And I'd like to eliminate the Date column just for this one subscription, because version numbers are more important.

**Comment by josh64x2** [comment]
@rcaputo can you think of a way that you would implement it? 
I would probably create a new table in the db, that maps a folder_id (a single feed is also a "folder"), to a bunch of attributes such as sort order, layout, style.
What do you think?

**Comment by rcaputo** [comment]
This is outside my comfort zone, but YOLO.  I just downloaded the source and built Vienna.  It was surprisingly straightforward, so thanks for that.

So, (folder_id, attribute, value)?  Whatever needs the value of a configuration "attribute" would walk up the Folders.parent_id chain starting at the current folder_id and stopping when the configuration value is found.  If parent_id reaches -1 without anything being found, the global Preferences are used.

That could work.  It seems kind of slow, but values could be cached so that lookups eventually needn't traverse the folder tree.  A change at any given folder_id could invalidate childrenor recalculate children in the cache.
